lib/auth/auth_with_roles.go

```diff
@@ -3144,6 +3144,25 @@ func (a *ServerWithRoles) GetSessionEvents(namespace string, sid session.ID, aft
 		return nil, trace.Wrap(err)
 	}
 
+	// emit a session recording view event for the audit log
+	user, err := a.GetCurrentUser(context.Background())
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	if a.authServer.emitter.EmitAuditEvent(a.authServer.closeCtx, &apievents.SessionRecordingAccess{
```

Comment:
Interestingly, as we emit these audit logs before we attempt to fetch the event, the audit log message will show even if the user tries to read an event that does not exist. I think that makes sense, but worth thinking about. Perhaps some kind of status in the audit event as to if the read succeeded or not would be valuable.